File: android/src/main/java/com/lightningkite/khrysalis/Image.actual.kt
package com.lightningkite.khrysalis

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import android.provider.MediaStore
import com.lightningkite.khrysalis.net.HttpClient
import com.lightningkite.khrysalis.rx.forever
import okhttp3.Call
import okhttp3.Callback
import okhttp3.Request
import okhttp3.Response
import java.io.IOException
import java.io.InputStream
import kotlin.math.max
import kotlin.math.min

/* SHARED DECLARATIONS
typealias Bitmap = Bitmap
 */

@Deprecated("Use Rx Style instead")
fun loadImage(image: Image, onResult: (Bitmap?) -> Unit) {
    image.load().subscribe { result, fail ->
        onResult(result)
    }.forever()
}

@Deprecated("Use Rx Style instead")
fun Image.load(onResult: (Bitmap?) -> Unit) {
    load().subscribe { result, fail ->
        onResult(result)
    }.forever()
}

File: ios/Khrysalis/src/Image.actual.swift
//Stub file made with Khrysalis 2 (by Lightning Kite)
import Foundation
import UIKit


//--- Bitmap
public typealias Bitmap = UIImage

//--- loadImage(Image, (Bitmap?)->Unit)
public func loadImage(_ image: Image, _ onResult: @escaping (Bitmap?) -> Void) -> Void {
    image.load().subscribe(
        onSuccess: { onResult($0) },
        onError: { onResult(nil) }
    )
}
public func loadImage(image: Image, onResult: @escaping (Bitmap?) -> Void) -> Void {
    return loadImage(image, onResult)
}

//--- Image.load((Bitmap?)->Unit)
public extension Image {
    func load(_ onResult: @escaping (Bitmap?) -> Void) -> Void {
        return loadImage(self, onResult)
    }
}
